### PR TITLE
Onboarding: New getting started panel

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Updates design and look and feel on the Getting started panel

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by jessover9000** [comment]
It's already looking quite nice, great work @peterholmberg 🙌 

Some details that I've noticed:

- Can we have the "Basic"/"advanced" intro text section vertically align centered?
- The colored border-top in the cards should be 2px instead of 1px.
- The chevron button is a bit hard to see, especially in dark theme. Can we have the chevron be bigger? 
- Currently the top part and the bottom "docs link" part link to the same thing. The top part should, however, link to the same pages as the old getting started panel (e.g. "Add data source" page).
- Links for the "Need help?" section are still missing.
- I clicked on "Remove this panel" on my local home dashboard and it just disappeared and now I don't know how to get it back in case I want it back (can you even get it back?). I think there should be a double confirmation to remove that panel (could be a modal or inline like in the User Profile admin section).
- Regarding "getting back the getting started panel", not sure whether it might be smart to add the option to bring it back later somehow in case a user wants to see it again? @torkelo thoughts?

**Comment by jessover9000** [comment]
I like the split up look, good work @peterholmberg 👍 some remaining nitpicks:

- for the cards links, currently the top part and the bottom "docs link" part link to the same thing. The top part should, however, link to the same pages as the old getting started panel (e.g. "Add data source" page or "Create dashboard" page).

- spacing of the "Need help" links could be bigger, margin-right: 16px should fix it
- between the 770px and 992px breakpoints the "Welcome" headline is very close to the "Need help" line. maybe add a margin-bottom: 6px; to the headline there
- the left alignment/margin-left of "Welcome to Grafana" jumps between the breakpoints for larger screen sizes. I think it would be nice if it aligned to the "Basic/Advanced" area in the lower panel :)
- below 545px, the text in the top welcome bar is too close to the top. an additional margin-top: 8px for the title would make the text look more vertically centered (or you can just try and auto-center it using css magic ;D)
- the Grafana swirl icon on the "Grafana fundamentals" tutorial card gets cut off in smaller widths. maybe just hide it instead as soon as it gets cut off?
- the "Remove this panel" link should only have a padding-top of 12px and then a margin-bottom of 24px, so that it looks more equally positioned to the top right corner.
- using the same background graphic for both the top welcome bar and the cards area makes the repetition of the graphic quite obvious (the crossing orbital lines give it away). also, for smaller widths the background gets in the way of the "Basic/Advanced" text description. maybe we can use a different background for the lower area (will send an SVG to Peter). 

things I'm not sure about whether they can be fixed now but where I think they could be better:
- when I click the cards, they only switch to complete status after a refresh. I think it would be nice to have more immediate feedback that clicking the card "completes" them
- the scrolling div for the basic steps is obviously not super elegant. I originally imagined that the basic and advanced steps are all part of the same sliding mechanism instead of being split up into two "pages". But maybe it's just a quick solution that works for now and we can improve this later :)

**Comment by peterholmberg** [comment]
> * the "Remove this panel" link should only have a padding-top of 12px and then a margin-bottom of 24px, so that it looks more equally positioned to the top right corner.

If it has less padding it won't be clickable due to the panel menu bar that will cover it.


> things I'm not sure about whether they can be fixed now but where I think they could be better:
> 
> * when I click the cards, they only switch to complete status after a refresh. I think it would be nice to have more immediate feedback that clicking the card "completes" them

> * the scrolling div for the basic steps is obviously not super elegant. I originally imagined that the basic and advanced steps are all part of the same sliding mechanism instead of being split up into two "pages". But maybe it's just a quick solution that works for now and we can improve this later :)

Yes, that's fairly advanced and we might want to revisit this another time.

**Comment by jessover9000** [comment]
> If it has less padding it won't be clickable due to the panel menu bar that will cover it.

Good point, I didn't realize. Then maybe increase the margin-right by 16px so that the top and right margin look about the same?

Other than that, looks and works nicely now 💯

**Comment by jessover9000** [comment]
Oh, and one more thing. I think the top welcome bar has no border but the cards/tutorial part does have a border. I think it should be consistent with the rest of the panels = they should all have borders ;)

**Comment by torkelo** [comment]
This looking good! Great job @peterholmberg & @jessover9000  

the link for create users and teams is not working: 
https://grafana.com/tutorials/create-users-and-teams/#0 

Think that should index should be a 1

Would be great if all the links to the docs & tutorials had a ?utm_source=grafana_gettingstarted url parameters (this goes for links in the welcome banner as well). That way we can track how many use these links.

File: public/app/plugins/panel/gettingstarted/GettingStarted.tsx [hunk 1]
```diff
@@ -137,34 +79,114 @@ export class GettingStarted extends PureComponent<PanelProps, State> {
...
       </div>
     );
   }
 }
+
+const getStyles = stylesFactory(() => {
+  const { theme } = config;
+  const backgroundImage = theme.isDark
+    ? 'public/img/Onboarding_Panel_dark.svg'
+    : 'public/img/Onboarding_Panel_light.svg';
```
Comment (dprokop) [inline public/app/plugins/panel/gettingstarted/GettingStarted.tsx:118]: You don't have to put the images in the public dir anymore. You can put them next to the component and then:

```
import imageLight from './imageLight.svg';
import imageDark from './imageDark.svg';
// ...
// styles
background-image: url(${theme.isLight ? imageLight : imageDark});
```

Comment (peterholmberg) [reply public/app/plugins/panel/gettingstarted/GettingStarted.tsx:118]: Nice!

File: public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  );
+};
+
+const handleTutorialClick = (event: MouseEvent<HTMLAnchorElement>, card: Card) => {
+  event.preventDefault();
+  const isSet = store.get(card.key);
+  if (!isSet) {
+    store.set(card.key, true);
+  }
+  window.open(card.href, '_blank');
```
Comment (dprokop) [inline public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx:40]: Aren't ad blockers going to block this?

File: public/app/plugins/panel/gettingstarted/GettingStarted.tsx [hunk 3]
```diff
@@ -137,34 +79,114 @@ export class GettingStarted extends PureComponent<PanelProps, State> {
   };
 
   render() {
-    const { checksDone } = this.state;
+    const { checksDone, currentStep, steps } = this.state;
+
     if (!checksDone) {
       return <div>checking...</div>;
```
Comment (dprokop) [inline public/app/plugins/panel/gettingstarted/GettingStarted.tsx:85]: Getting this on a slow connection:

image

File: public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx (hunk 2 above)
Comment (peterholmberg) [reply public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx:40]: I don't know, are they?

Comment (torkelo) [reply public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx:40]: if you have a onClick & href & target="_blank" does the onClick trigger?

File: public/app/plugins/panel/gettingstarted/steps.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    title: 'Advanced',
+    info: ' Manage your users and teams and add plugins. These steps are optional',
+    done: false,
+    cards: [
+      {
+        type: 'tutorial',
+        heading: 'Users',
+        title: 'Create users and teams',
+        info: 'Learn to organize your users in teams and manage resource access and roles.',
+        href: 'https://grafana.com/tutorials/create-users-and-teams?utm_source=grafana_gettingstarted',
```
Comment (torkelo) [inline public/app/plugins/panel/gettingstarted/steps.ts:88]: can we add this "utm_source=grafana_gettingstarted" in the render step as well so it does not need to be part of every item

File: public/app/plugins/panel/gettingstarted/steps.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    info: 'The steps below will guide you to quickly finish setting up your Grafana installation.',
+    done: false,
+    cards: [
+      {
+        type: 'tutorial',
+        heading: 'Data source and dashboards',
+        title: step1TutorialTitle,
+        info:
+          'Set up and understand Grafana if you have no prior experience. This tutorial guides you through the entire process and covers the “Data source” and “Dashboards” steps to the right.',
+        href: 'https://grafana.com/tutorials/grafana-fundamentals?utm_source=grafana_gettingstarted',
```
Comment (torkelo) [inline public/app/plugins/panel/gettingstarted/steps.ts:32]: would be nice to add ?utm_source=grafana_gettingstarted in render if possible

**Review by torkelo — APPROVED** [review]
Great job

File: public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx (hunk 2 above)
Comment (peterholmberg) [reply public/app/plugins/panel/gettingstarted/components/TutorialCard.tsx]: It seems to work at least. But can remove the target prop from a element

**Comment by torkelo** [comment]
For macbook 13" there is a scrollbar, can we tweak the responsive rules to reduce the padding to make that look better? 

And adding utm_source=grafana_welcom for the links in the welcome banner would be great, after that merge it! 

Screenshot from 2020-05-12 16-25-01

**Pull request merged**